### ISSUE TITLE
Highlight doesn't work in the first change

### ISSUE THREAD

**Issue by medhatdawoud** [medhatdawoud/react-change-highlight#21]
**Which version?**
1.2.3

**Describe the bug**
when I wrap an element and give it a ref as required it does work after the first time but the first change is not highlighted.

**To Reproduce**
1. Wrap an element with `ChangeHighlight`
2. Add ref to the element
3. Change the content of that element using a state change
4. first time it wouldn't highlight

**Expected behavior**
It should highlight from the first change

**Desktop (please complete the following information):**
 - OS: osX 10.14.5
 - Browser: Chrome
 - Version: 78

**Comment by EslamHiko** [comment]
I can't reproduce the bug, it's working fine with me.

My code :
```javascript
class App extends React.Component {


  constructor (props) {
      super(props);
      this.state = {
          // modal should be closed on page load
          count: 0,
          count2:0
      };
      this.ref1 = React.createRef();
      this.ref2 = React.createRef();
      this.countUp = this.countUp.bind(this);
    }


  countUp(){
    this.setState({count:this.state.count+1,count2:this.state.count2+1})

  };
  componentDidMount (){
    this.countUp();
  }
  render(){
    return <div className="App">
      <button onClick={this.countUp}>Count up</button>
      <br />
      <ChangeHighlight showAfter={100} hideAfter={2000}>
        <h1 ref={this.ref1}>{this.state.count}</h1>
        <p>
          <div>
          <ChangeHighlight showAfter={3000} highlightStyle="red">
            <h1 ref={this.ref2}>{this.state.count2}</h1>
            </ChangeHighlight>
          </div>
        </p>
      </ChangeHighlight>
    </div>
  }
}

const rootElement = document.getElementById("root");
ReactDOM.render(<App />, rootElement);
```

**Comment by medhatdawoud** [comment]
Try to make a more complex example, like creating 2 custom components inside each other and wrap them in more than one layer of elements then wrap the root one with `ChangeHighlight` and put the ref on the grand child

**Comment by mohamedGamalAbuGalala** [comment]
@medhatdawoud @EslamHiko 
Tried to produce the bug with many test cases, only one case is producing the bug.

**Example Code**
```

const App: React.FC = () => {
  const [state, handleChange] = useForm({
    result1: ""
  });
  return (
    <div className="App">
      <ChangeHighlight showAfter={10}>
        <p ref={React.createRef()}>{state.result1}</p>
      </ChangeHighlight>

      <input
        type="text"
        name="result1"
        value={state.result1}
        onChange={handleChange}
      />
    </div>
  );
};
```

**Scenario**
* result1 is empty so the first time it just add it to shadow dom and after can highlight it.

**Comment by medhatdawoud** [comment]
@mohamedGamalAbuGalala good that you could reproduce it, if possible can you make that example on a codesandbox so we can check and even after fixing we can try it out without pulling and building. Thanks

**Comment by mohamedGamalAbuGalala** [comment]
Okay, I will tomorrow 👌

Mohamed Gamal

**Comment by mohamedGamalAbuGalala** [comment]
https://codesandbox.io/s/wizardly-hawking-0l470?fontsize=14&hidenavigation=1&theme=dark

You can just build and then replace files in the `ChangeHighlight` folder.

In this sample, there is a lot of bugs but I think that all is related to #20 
What do you think? :thinking:

**Comment by medhatdawoud** [comment]
It's fine but actually we need one specific codesandbox for this specific issue only, click and go :)

**Comment by mohamedGamalAbuGalala** [comment]
@medhatdawoud 
You misunderstand me, I mean this issue will be solved when we solve #20, cuz this issue is about detecting newly added elements.
